src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -478,12 +478,6 @@ private static CastResult TryGet(nuint source, nuint target)
         {
             MethodTable* mt = RuntimeHelpers.GetMethodTable(obj);
 
```

Comment:
Could you please add back the assert that was here before? 

https://github.com/dotnet/runtime/pull/69869/files#diff-996cab96d226ffc0be340a84d4a583c9b8d05e443a90d98e0352c948af57291dL480